lib/spot-checker.js

```diff
@@ -113,36 +120,20 @@ export default class Spark {
       if (res.ok) {
         resetTimeout()
         for await (const value of res.body) {
-          if (stats.firstByteAt === null) {
-            stats.firstByteAt = new Date()
-          }
           stats.byteLength += value.byteLength
 
-          // We want to limit how large content we are willing to download.
-          // 1. To make sure we don't spend too much time (and network bandwidth) on a single task,
-          //    so that we can complete more tasks per round
-          // 2. Until we have streaming hashes, we need to keep the entire payload in memory, and so
-          //    we need to put an upper limit on how much memory we consume.
-          if (stats.byteLength > MAX_CAR_SIZE) {
-            stats.carTooLarge = true
-            break
-          }
-
           const offset = carBuffer.byteLength
           carBuffer.resize(offset + value.byteLength)
           carBytes.set(value, offset)
 
           resetTimeout()
         }
 
-        if (!stats.carTooLarge) {
-          await verifyContent(cid, carBytes)
-
-          const digest = await crypto.subtle.digest('sha-256', carBytes)
-          // 12 is the code for sha2-256
-          // 20 is the digest length (32 bytes = 256 bits)
-          stats.carChecksum = '1220' + encodeHex(digest)
-        }
+        await verifyContent(cid, carBytes)
+        const digest = await crypto.subtle.digest('sha-256', carBytes)
+        // 12 is the code for sha2-256
+        // 20 is the digest length (32 bytes = 256 bits)
+        stats.carChecksum = '1220' + encodeHex(digest)
```

Comment:
What would you like to use `carChecksum` created by the spot-checker for? Can we remove it from the initial version to keep it simpler?

Reply:
I agree, it has been removed it. We're also not collecting the file bytes into buffer anymore so there's an additional reason.